src/EventStore.Core/Services/Gossip/GossipServiceBase.cs

```diff
@@ -94,31 +104,34 @@ public abstract class GossipServiceBase : IHandle<SystemMessage.SystemInit>,
 		}
 
 		public void Handle(GossipMessage.GotGossipSeedSources message) {
-			var now = DateTime.UtcNow;
+			var now = _timeProvider.UtcNow;
 			var dnsCluster = new ClusterInfo(
 				message.GossipSeeds.Select(x => MemberInfo.ForManager(Guid.Empty, now, true, x, x)).ToArray());
 
 			var oldCluster = _cluster;
-			_cluster = MergeClusters(_cluster, dnsCluster, null, x => x);
+			_cluster = MergeClusters(_cluster, dnsCluster, null, x => x, _timeProvider.UtcNow, NodeInfo, CurrentMaster,
+				AllowedTimeDifference, DeadMemberRemovalTimeout);
 			LogClusterChange(oldCluster, _cluster, null);
 
 			_state = GossipState.Working;
-			Handle(new GossipMessage.Gossip(0)); // start gossiping
+			Handle(new GossipMessage.Gossip(0));
 		}
 
 		public void Handle(GossipMessage.Gossip message) {
 			if (_state != GossipState.Working)
 				return;
 
-			var node = GetNodeToGossipTo(_cluster.Members);
+			var node = _getNodeToGossipTo(_cluster.Members);
 			if (node != null) {
-				_cluster = UpdateCluster(_cluster, x => x.InstanceId == NodeInfo.InstanceId ? GetUpdatedMe(x) : x);
+				_cluster = UpdateCluster(_cluster, x => x.InstanceId == NodeInfo.InstanceId ? GetUpdatedMe(x) : x,
+					_timeProvider, DeadMemberRemovalTimeout);
 				_bus.Publish(new HttpMessage.SendOverHttp(node.InternalHttpEndPoint,
-					new GossipMessage.SendGossip(_cluster, NodeInfo.InternalHttp), DateTime.Now.Add(GossipInterval)));
+					new GossipMessage.SendGossip(_cluster, NodeInfo.InternalHttp),
+					_timeProvider.LocalTime.Add(GossipInterval)));
 			}
 
-			var interval = message.GossipRound < 20 ? GossipStartupInterval : GossipInterval;
-			var gossipRound = Math.Min(2000000000, node == null ? message.GossipRound : message.GossipRound + 1);
+			var interval = message.GossipRound < GossipRoundStartupThreshold ? GossipStartupInterval : GossipInterval;
+			var gossipRound = Math.Min(int.MaxValue, node == null ? message.GossipRound : message.GossipRound + 1);
```

Comment:
```suggestion
			var gossipRound = Math.Min(int.MaxValue - 1, node == null ? message.GossipRound : message.GossipRound + 1);
```

nit pick: this should be int.MaxValue - 1, otherwise if `message.GossipRound` has a value of `int.maxValue`, `message.GossipRound + 1` will overflow.